Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -34,4 +41,15 @@ def env_apple
   def oracle_java_os
     :darwin
   end
+
+  JAVA_CASK_MAP = {
+    "1.6" => "caskroom/versions/java6",
+    "1.7" => "caskroom/versions/zulu7",
+    "1.8" => "caskroom/versions/java8",
+    "9.0" => "caskroom/cask/java",
+  }.freeze
+
+  def legacy_java_cask(cask_version = "1.8")
+    @cask = JAVA_CASK_MAP.fetch(cask_version.to_s)
```

Comment:
Feels a bit weird to have `cask "java"` and setting this here, too? Also, `caskroom/cask/java` doesn't agree with the version above.

Reply:
Looking at this now, the reasoning I had for having `"1.8"` as the default `cask_version` makes no sense.

Combined with your comment about limiting it to `Java 8` and not setting it in two places, I am considering refactoring this to completely replace setting the `cask` variable, even for the default case when a version tag isn't passed.